### DISCUSSION TITLE
Current Visitor

### DISCUSSION THREAD

**Discussion by YanDevDe** [plausible/analytics#1592] [Anything Else]
Hello!

Is there a way to keep "current visitor" even if I'm leaving website open for a long time period?
The reason is because I'm using this in an electron software and it tends to be opened for a very long time period. For this reason, "current visitor" will be dropped back again after engaging the software.

Perhaps having both options, "current engaging visitors" and "current visitors" would be fancy to have?

Workaround would be for me to use interval to send event to keep "current visitor" alive, if so, what is the suitable interval time?

Thanks!

- Yan

**Comment by ukutaht** [comment]
Hey @YanDevDe and thanks for the suggestion.

Keeping the visitor 'alive' would require some sort of pinging like you said. Now, this will have it's own problems. I think if you just implement it with `setInterval` it will also ping if the window is minimized. So you might want to do something fancy with `visibilityState` and/or user input events if you really want to figure out if the visitor is engaged.

> Workaround would be for me to use interval to send event to keep "current visitor" alive, if so, what is the suitable interval time?

Yes you can send a custom event to keep the visitor 'active'.  Keep in mind the caveats above. 'Current visitors' are defined as visitors who triggered an event in the last 5 minutes so you'll need to ping at a similar interval.
